### ISSUE TITLE
Double completion of macros when they are typed completely

### ISSUE THREAD

**Issue by weirdsmiley** [rust-analyzer/rust-analyzer#9904]
I am using CoC with Vim. The completions work fine when the macro is incomplete. But when the macro is complete (including the last `!` character) and then completion is triggered, then it completes the macro twice.

An example is:

Before completion:
```rust
fn main() {
    println!<|>
}
```

After completion:
```rust
fn main()
    println!println!(<|>)
}
```

**Comment by lnicola** [comment]
This happens in VS Code too.

**Comment by Veykril** [comment]
I imagine this might be due to `!` not being in the trigger character set?
